src/routes/transactions/converters/details.rs

```diff
@@ -177,21 +180,18 @@ pub async fn is_trusted_delegate_call(
     to: &str,
     data_decoded: &Option<DataDecoded>,
     info_provider: &(impl InfoProvider + Sync),
-) -> Option<bool> {
+) -> ApiResult<Option<bool>> {
     if operation == &Operation::DELEGATE {
-        info_provider
-            .contract_info(to)
-            .await
-            .map(|contract_info| {
-                // In the case of a known `multiSend` method call, we can still have internal transactions with DELEGATE calls
-                // which is why we only check when `trusted_for_delegate_call = true`
-                contract_info.trusted_for_delegate_call
-                    && !data_decoded
-                        .as_ref()
-                        .map_or(false, |data_decoded| data_decoded.has_nested_delegated())
-            })
-            .ok()
+        let contract_info = info_provider.contract_info(to).await?;
+
+        let has_nested_delegate_calls = !data_decoded
+            .as_ref()
+            .map_or(false, |data_decoded| data_decoded.has_nested_delegated());
+
+        let is_trusted_delegate_call =
+            contract_info.trusted_for_delegate_call && has_nested_delegate_calls;
```

Comment:
The way that it was written before was short-circuiting to prevent the http request in case `contract.trusted_for_delegate_call` was `true`. I don't mind the change, I am just pointing out a big change in behaviour.

Reply:
Never mind this comment